File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.cpp
class Solution {
public:
    vector<int> getSumAbsoluteDifferences(vector<int>& nums) {
        int n = nums.size();
        vector<int>ans(n, 0);
        
        for(int i = 1; i < n; i++) {
            ans[0]+=(nums[i]-nums[0]);
        }
        
        for(int j = 1;j < n; j++) {
            ans[j] = ans[j-1] + (nums[j] - nums[j-1]) * j - (nums[j] - nums[j-1]) * (n-j);
        }
        
        return ans;
    }
};

File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.java
class Solution {
    public int[] getSumAbsoluteDifferences(int[] nums) {
        int n = nums.length;
        int result[] = new int[n];
        
        for(int i = 1; i < n; i++) {
            result[0] += (nums[i] - nums[0]);
        }
        
        for(int j = 1; j < n; j++) {
            result[j] = result[j - 1] + (nums[j] - nums[j - 1]) * j - (nums[j] - nums[j - 1]) * (n - j);
        }
        
        return result;        
    }
}